system_tests/datastore.py

```diff
@@ -41,18 +44,22 @@ class Config(object):
     CLIENT = None
 
 
+def clone_client(client):
+    # Fool the Client constructor to avoid creating a new connection.
+    cloned_client = datastore.Client(project=client.project,
+                                     namespace=client.namespace,
+                                     http=object())
+    cloned_client.connection = client.connection
+    return cloned_client
+
+
 def setUpModule():
     if EMULATOR_DATASET is None:
-        client.DATASET = TESTS_DATASET
-        Config.CLIENT = datastore.Client()
+        client_mod.DATASET = TESTS_DATASET
+        Config.CLIENT = datastore.Client(namespace=TEST_NAMESPACE)
     else:
-        Config.CLIENT = datastore.Client(project=EMULATOR_DATASET)
-        populate_datastore.add_characters(client=Config.CLIENT)
-
-
-def tearDownModule():
-    if EMULATOR_DATASET is not None:
-        clear_datastore.remove_all_entities(client=Config.CLIENT)
+        Config.CLIENT = datastore.Client(project=EMULATOR_DATASET,
+                                         namespace=TEST_NAMESPACE)
```

Comment:
I can see that now.  I guess I mis-read the diff:  the lack of parallelism between the two clauses might be what threw me.

LGTM